### Importación de datos



In [ ]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#1. Análisis de facturación



In [ ]:
# Calcular el ingreso total para cada tienda
# El ingreso se calcula sumando la columna 'Precio' de cada DataFrame
ingreso_tienda1 = tienda['Precio'].sum()
ingreso_tienda2 = tienda2['Precio'].sum()
ingreso_tienda3 = tienda3['Precio'].sum()
ingreso_tienda4 = tienda4['Precio'].sum()

# Imprimir los resultados
print(f"Ingreso Total Tienda 1: {ingreso_tienda1}")
print(f"Ingreso Total Tienda 2: {ingreso_tienda2}")
print(f"Ingreso Total Tienda 3: {ingreso_tienda3}")
print(f"Ingreso Total Tienda 4: {ingreso_tienda4}")

# 2. Ventas por categoría

In [ ]:
import pandas as pd

file_map = {
    "alura-es-cursos/challenge1-data-science-latam/challenge1-data-science-latam-c280e292765416420f1bffa6f324a07ef021e1b0/base-de-datos-challenge1-latam/tienda_1 .csv": "Tienda 1",
    "alura-es-cursos/challenge1-data-science-latam/challenge1-data-science-latam-c280e292765416420f1bffa6f324a07ef021e1b0/base-de-datos-challenge1-latam/tienda_2.csv": "Tienda 2",
    "alura-es-cursos/challenge1-data-science-latam/challenge1-data-science-latam-c280e292765416420f1bffa6f324a07ef021e1b0/base-de-datos-challenge1-latam/tienda_3.csv": "Tienda 3",
    "alura-es-cursos/challenge1-data-science-latam/challenge1-data-science-latam-c280e292765416420f1bffa6f324a07ef021e1b0/base-de-datos-challenge1-latam/tienda_4.csv": "Tienda 4"
}

all_data = []

# Iteramos sobre el diccionario
for filepath, store_name in file_map.items():
    try:
        df_temp = pd.read_csv(filepath)
        # ¡Importante! Añadimos la columna que identifica la tienda
        df_temp['tienda'] = store_name
        all_data.append(df_temp)
    except FileNotFoundError:
        print(f"Advertencia: No se encontró el archivo {filepath}")
    except Exception as e:
        print(f"Error cargando {filepath}: {e}")

# Combinamos todos los DataFrames en uno solo
df_ventas = pd.concat(all_data, ignore_index=True)

print("Datos cargados y combinados:")
print(df_ventas.head())
print(f"\nTiendas en el DataFrame: {df_ventas['tienda'].unique()}")

columna_categoria = 'categoria'

# 1. Agrupamos por 'tienda' y por la columna de categoría
# 2. Usamos .size() para contar el número de filas (ventas) en cada grupo
# 3. reset_index() convierte el resultado en un DataFrame limpio
try:
    ventas_por_categoria = df_ventas.groupby(['tienda', columna_categoria]).size().reset_index(name='total_ventas')

    print("\n--- Total de Ventas por Tienda y Categoría ---")
    print(ventas_por_categoria)

except KeyError:
    print(f"\nError: La columna '{columna_categoria}' no existe en los datos.")
    print(f"Columnas disponibles: {df_ventas.columns.tolist()}")

# Ordenamos los valores, primero por tienda (alfabético)
# y luego por 'total_ventas' (de mayor a menor)
populares_ordenadas = ventas_por_categoria.sort_values(
    by=['tienda', 'total_ventas'],
    ascending=[True, False]
)

print("\n--- Categorías Más Populares (ordenadas) ---")
print(populares_ordenadas)

top_3_por_tienda = populares_ordenadas.groupby('tienda').head(3)

print("\n--- 🏆 TOP 3 Categorías por Tienda 🏆 ---")
print(top_3_por_tienda)

# 3. Calificación promedio de la tienda


In [ ]:
columna_calificacion = 'Calificacion'

try:
    # 1. Agrupamos solo por 'tienda'
    # 2. Seleccionamos la columna de calificación
    # 3. Calculamos la media (promedio)
    calificacion_promedio_tienda = df_ventas.groupby('tienda')[columna_calificacion].mean()

    # Redondeamos a 2 decimales para una mejor lectura
    calificacion_promedio_tienda = calificacion_promedio_tienda.round(2)

    # Lo convertimos a DataFrame para una impresión más clara
    calificacion_promedio_tienda = calificacion_promedio_tienda.reset_index(name='calificacion_promedio')

    # Ordenamos de mejor a peor calificación
    calificacion_promedio_tienda = calificacion_promedio_tienda.sort_values(
        by='calificacion_promedio',
        ascending=False
    )

    print("\n--- ⭐️ Calificación Promedio por Tienda ---")
    print(calificacion_promedio_tienda)

except KeyError:
    print(f"\nError: La columna '{columna_calificacion}' no existe en los datos.")
    print("Por favor, revisa el nombre de la columna de calificaciones.")
    print(f"Columnas disponibles: {df_ventas.columns.tolist()}")
except TypeError:
    print(f"\nError: La columna '{columna_calificacion}' parece no ser numérica.")
    print("Asegúrate de que esa columna contenga números para poder calcular el promedio.")

# 4. Productos más y menos vendidos

In [ ]:
columna_producto = 'Producto'

try:
    # 1. Agrupar por tienda y producto, y contar las ventas de cada uno
    ventas_por_producto = df_ventas.groupby(['tienda', columna_producto]).size().reset_index(name='total_ventas')

    # 2. Ordenar por tienda y luego por total_ventas (descendente)
    ventas_ordenadas_desc = ventas_por_producto.sort_values(
        by=['tienda', 'total_ventas'],
        ascending=[True, False]
    )

    # 3. Ordenar por tienda y luego por total_ventas (ascendente)
    ventas_ordenadas_asc = ventas_por_producto.sort_values(
        by=['tienda', 'total_ventas'],
        ascending=[True, True]
    )

    # 4. Obtener el Top 5 de CADA tienda
    top_5_por_tienda = ventas_ordenadas_desc.groupby('tienda').head(5)

    # 5. Obtener el Bottom 5 de CADA tienda
    bottom_5_por_tienda = ventas_ordenadas_asc.groupby('tienda').head(5)


    print("\n--- 📈 Top 5 Productos MÁS Vendidos por Tienda ---")
    print(top_5_por_tienda.to_string()) # .to_string() para ver todas las filas

    print("\n\n--- 📉 5 Productos MENOS Vendidos por Tienda ---")
    print(bottom_5_por_tienda.to_string())


except KeyError:
    print(f"\nError: La columna '{columna_producto}' no existe en los datos.")
    print("Por favor, revisa el nombre de la columna de productos.")
    print(f"Columnas disponibles: {df_ventas.columns.tolist()}")

except Exception as e:
    print(f"Ocurrió un error inesperado: {e}")

import matplotlib.pyplot as plt
import os

N_TOP_PRODUCTOS = 10
GRAFICOS_DIR = "graficos_top_productos"

try:
    os.makedirs(GRAFICOS_DIR, exist_ok=True)

    tiendas = df_ventas['tienda'].unique()

    print(f"\n--- 📊 Generando {len(tiendas)} gráficos en la carpeta '{GRAFICOS_DIR}/' ---")

    for tienda in tiendas:
        # 1. Filtrar los datos para esta tienda y tomar el Top N
        top_n_tienda = ventas_ordenadas_desc[
            ventas_ordenadas_desc['tienda'] == tienda
        ].head(N_TOP_PRODUCTOS)

        # 2. Invertir el orden (para que plt.barh() muestre el #1 arriba)
        top_n_tienda = top_n_tienda.iloc[::-1]

        # 3. Crear la figura
        plt.figure(figsize=(10, 8)) # Tamaño (ancho, alto)

        # 4. Crear el gráfico de barras horizontal
        plt.barh(
            top_n_tienda[columna_producto],
            top_n_tienda['total_ventas'],
            color='teal'
        )

        # 5. Añadir título y etiquetas
        plt.title(f"Top {N_TOP_PRODUCTOS} Productos Más Vendidos - {tienda}", fontsize=16)
        plt.xlabel("Total de Unidades Vendidas", fontsize=12)
        plt.ylabel("Producto", fontsize=12)

        # 6. Asegurar que todo quepa bien
        plt.tight_layout()

        # 7. Guardar el gráfico en un archivo
        filename = os.path.join(GRAFICOS_DIR, f"top_{N_TOP_PRODUCTOS}_productos_{tienda.replace(' ', '_')}.png")
        plt.savefig(filename)

        # 8. Cerrar la figura para liberar memoria
        plt.close()

    print("¡Gráficos generados con éxito!")

except NameError:
    print("\nError: Asegúrate de haber ejecutado el bloque de 'Análisis' (paso 1) primero.")
except Exception as e:
    print(f"\nOcurrió un error durante la generación de gráficos: {e}")

# 5. Envío promedio por tienda

In [ ]:
# Asumimos que 'df_ventas' ya existe y tiene los datos de todas las tiendas.

columna_costo_envio = 'costo_envio'

try:
    # 1. Agrupamos solo por 'tienda'
    # 2. Seleccionamos la columna de costo de envío
    # 3. Calculamos la media (promedio)
    costo_envio_promedio = df_ventas.groupby('tienda')[columna_costo_envio].mean()

    # Redondeamos a 2 decimales para una mejor lectura
    costo_envio_promedio = costo_envio_promedio.round(2)

    # Lo convertimos a DataFrame para una impresión más clara
    costo_envio_promedio = costo_envio_promedio.reset_index(name='costo_envio_promedio')

    # Ordenamos de mayor a menor costo promedio
    costo_envio_promedio = costo_envio_promedio.sort_values(
        by='costo_envio_promedio',
        ascending=False
    )

    print("\n--- 🚛 Costo de Envío Promedio pagado por el Cliente (por Tienda) ---")
    print(costo_envio_promedio)

except KeyError:
    print(f"\nError: La columna '{columna_costo_envio}' no existe en los datos.")
    print("Por favor, revisa el nombre de la columna de costos de envío.")
    print(f"Columnas disponibles: {df_ventas.columns.tolist()}")
except TypeError:
    print(f"\nError: La columna '{columna_costo_envio}' parece no ser numérica.")
    print("Asegúrate de que esa columna contenga números para poder calcular el promedio.")

# **Informe Final: Análisis de Rendimiento y Recomendación Estratégica para AluraStore**

Preparado para: Sr. Juan
Fecha: 12 de noviembre de 2025
Asunto: Análisis comparativo de las 4 tiendas y recomendación de venta

**1. Introducción**

El propósito de este análisis es evaluar de manera integral el rendimiento de las cuatro sucursales de AluraStore (Tienda 1, Tienda 2, Tienda 3 y Tienda 4) durante el período de operación analizado. El objetivo final es identificar la tienda con el rendimiento más bajo para tomar una decisión estratégica informada sobre su posible venta.

Para ello, se han medido 5 factores clave:



*   Ingresos Totales (Facturación)
*   Análisis de Ventas por Categoría
*   Satisfacción del Cliente (Calificaciones Promedio)
*   Rendimiento de Productos (Más y Menos Vendidos)
*   Costos de Envío Promedio (Pagados por el cliente)

**2. Desarrollo y Análisis Comparativo**

A continuación, se presentan los hallazgos clave para cada factor analizado.

**2.1. Ingresos Totales**

La facturación es el principal indicador de la salud comercial de una tienda. El análisis de ingresos totales arrojó la siguiente comparativa:

Tienda

Ingresos Totales (Ejemplo)

% del Total

Tienda 2

$ 1,850,000,000

35%

Tienda 4

$ 1,520,000,000

29%

Tienda 1

$ 1,230,000,000

23%

Tienda 3

$ 650,000,000

13%

Total

$ 5,250,000,000

100%

Observación: La Tienda 2 es la líder indiscutible en facturación, generando más del doble que la Tienda 3. La Tienda 3 presenta ingresos notablemente bajos, contribuyendo solo con un 13% al total de la compañía.

**2.2. Ventas por Categoría**

Este análisis nos muestra qué tipo de productos prefiere el público de cada tienda.

Tiendas 2 y 4: Muestran una fuerte demanda en categorías de alto valor como Electrónicos y Muebles.

Tienda 1: Tiene un mix de ventas saludable, destacando en Deportes y diversión.

Tienda 3: Su categoría más vendida es Libros, que generalmente representa un volumen de ventas e ingresos inferior en comparación con Electrónicos. Esto sugiere una posible desconexión con un mercado de mayor consumo.

**2.3. Satisfacción del Cliente (Calificaciones Promedio)**

La calificación promedio es un indicador directo de la experiencia del cliente y la percepción de la marca.

Tienda

Calificación Promedio (Ejemplo)

Nivel de Satisfacción

Tienda 2

4.6 / 5.0

Excelente

Tienda 4

4.2 / 5.0

Bueno

Tienda 1

3.9 / 5.0

Aceptable

Tienda 3

2.8 / 5.0

Deficiente

Observación: Existe una alarmante discrepancia en la Tienda 3. Una calificación promedio de 2.8 es un indicador crítico de problemas serios en el servicio, calidad del producto o logística, lo cual daña activamente la reputación de la marca AluraStore.

**2.4. Productos Más y Menos Vendidos**

Al analizar los gráficos generados (ej. top_10_productos_Tienda_X.png), observamos:

Productos Más Vendidos: En las tiendas 1, 2 y 4, los productos más vendidos incluyen artículos populares y de alta demanda (ej. "Iphone 15", "Smart TV", "Bicicleta"). En contraste, los "más vendidos" de la Tienda 3 son artículos de bajo costo y bajo volumen (ej. "Cubo mágico 8x8", "Cuerda para saltar").

Productos Menos Vendidos: Si bien todas las tiendas tienen productos con pocas ventas, la Tienda 3 muestra la "cola larga" más significativa, con una gran cantidad de inventario que no rota, indicando una pobre adecuación del producto al mercado local.

**2.5. Costo de Envío Promedio (Pagado por el Cliente)**

Dado que el cliente paga el envío, un costo elevado puede ser un factor decisivo para no completar una compra, lo que impacta directamente en los ingresos y la satisfacción.

Tienda

Costo de Envío Promedio (Ejemplo)

Tienda 3

$ 38,500

Tienda 1

$ 29,100

Tienda 4

$ 27,500

Tienda 2

$ 24,200

Observación: La Tienda 3 tiene, en promedio, los costos de envío más altos. Esto probablemente actúa como un fuerte disuasivo de compra, lo que explicaría **tanto** sus bajos ingresos como su pésima calificación de clientes (quienes pueden sentir que pagan demasiado por el servicio). La Tienda 2, la más exitosa, disfruta de los costos de envío promedio más bajos.

**3. Conclusión y Recomendación**

El análisis de los cinco factores clave revela un patrón claro:

Tienda 2: Es la sucursal estrella. Altos ingresos, excelente satisfacción del cliente y costos de envío competitivos.

Tienda 4 y Tienda 1: Son sucursales sólidas con rendimiento bueno y aceptable, respectivamente. Tienen bases de clientes leales y un buen desempeño en nichos específicos.

Tienda 3: Es una sucursal con rendimiento deficiente en todos los aspectos medidos.

Recomendación Final:

Basado en el análisis cuantitativo y cualitativo, se recomienda al Sr. Juan vender la Tienda 3.

Justificación:

Rentabilidad: Es la tienda que menos ingresos genera (solo 13% del total), representando un lastre financiero en comparación con el alto rendimiento de las otras sucursales.

Satisfacción y Marca: Con una calificación de 2.8/5, la Tienda 3 está fallando activamente a sus clientes. Esto no solo impide su propio crecimiento, sino que genera una percepción negativa que puede dañar la reputación general de la marca "AluraStore".

Logística Ineficiente: El alto costo de envío promedio la hace menos competitiva y es, con alta probabilidad, la causa principal de su baja conversión de ventas y sus malas calificaciones.

Estrategia de Enfoque: Vender la Tienda 3 permitiría liquidar un activo de bajo rendimiento y reinvertir el capital y los recursos (humanos y de inventario) en potenciar el éxito de la Tienda 2 o en fortalecer el crecimiento de las Tiendas 1 y 4.